packages/frontend/editor-ui/src/features/ai/chatHub/composables/useChatCredentials.ts

```diff
@@ -90,13 +92,27 @@ export function useChatCredentials(userId: string) {
 		selectedCredentials.value = { ...selectedCredentials.value, [provider]: id };
 	}
 
-	onMounted(async () => {
-		await Promise.all([
-			credentialsStore.fetchCredentialTypes(false),
-			credentialsStore.fetchAllCredentials(),
-		]);
-		isInitialized.value = true;
-	});
+	watch(
+		() => projectStore.personalProject,
+		async (personalProject) => {
+			if (personalProject) {
+				const hasGlobalCredentialRead = hasPermission(['rbac'], {
+					rbac: { scope: 'credential:read' },
+				});
+
+				await Promise.all([
+					credentialsStore.fetchCredentialTypes(false),
+					// For non-owner users only fetch credentials from personal project.
```

Comment:
How does this condition work? I tried my secondary account in local instance and it also has `credential:read` scope.

Reply:
I think it's "member" (2nd row in the screenshot)

<img width="1245" height="527" alt="Screenshot 2025-11-25 at 10 50 28" src="https://github.com/user-attachments/assets/3f81cc14-9028-44bd-af03-4eadcea7a0c6" />